src/Microsoft.NetCore.Analyzers/Core/Security/SystemSecurityCryptographyResources.resx

```diff
@@ -222,6 +222,15 @@
   <data name="DoNotUseMD5Description" xml:space="preserve">
     <value>This type implements MD5, a cryptographically insecure hashing function. Hash collisions are computationally feasible for the MD5 and HMACMD5 algorithms. Replace this usage with a SHA-2 family hash algorithm (SHA512, SHA384, SHA256).</value>
   </data>
+  <data name="DoNotUseObsoleteKDFAlgorithm" xml:space="preserve">
+    <value>Do Not Use Obsolete KDF Algorithm</value>
+  </data>
+  <data name="DoNotUseObsoleteKDFAlgorithmDescription" xml:space="preserve">
+    <value>Password-based Key Derivation should use PBKDF2 with SHA2.  Avoid using PasswordDeriveBytes since it's generating PBKDF1 key. Avoid using CryptDeriveBytes since it doesn't use iteration count, salt and IV.</value>
```

Comment:
>Password-based Key Derivation should use PBKDF2 with SHA2.  Avoid using PasswordDeriveBytes since it's generating PBKDF1 key. Avoid using CryptDeriveBytes since it doesn't use iteration count, salt and IV. [](start = 11, length = 205)

Password-based key derivation should use PBKDF2 with SHA-2. Avoid using PasswordDeriveBytes since it generates a PBKDF1 key. Avoid using CryptDeriveBytes since it doesn't use iteration count, salt and initialization vector.

Reply:
Sorry, one more edit.  I don't see a CryptDeriveBytes, maybe that should say CryptDeriveKey?  And that looks like it has an initialization vector parameter.
Password-based key derivation should use PBKDF2 with SHA-2. Avoid using PasswordDeriveBytes since it generates a PBKDF1 key. Avoid using CryptDeriveKey since it doesn't use the iteration count or salt.

---
In reply to: [274694245](https://github.com/dotnet/roslyn-analyzers/pull/2298#discussion_r274694245) [](ancestors = 274694245)